src/include/sof/audio/component_ext.h

```diff
@@ -147,13 +148,46 @@ static inline int comp_trigger_remote(struct comp_dev *dev, int cmd)
 	return idc_send_msg(&msg, IDC_BLOCKING);
 }
 
+static inline int comp_trigger_local(struct comp_dev *dev, int cmd)
+{
+	int ret;
+
+	ret = dev->drv->ops.trigger(dev, cmd);
+
+	/* start a thread in case of shared component or DP scheduling */
+	if (dev->task) {
+		/* schedule or cancel task, component must be shared or in DP domain */
+		assert(dev->is_shared ||
+		       dev->ipc_config.proc_domain == COMP_PROCESSING_DOMAIN_DP);
+
+		switch (cmd) {
+		case COMP_TRIGGER_START:
+		case COMP_TRIGGER_RELEASE:
+			schedule_task(dev->task, 0, dev->period);
+			break;
+		case COMP_TRIGGER_XRUN:
+		case COMP_TRIGGER_PAUSE:
+		case COMP_TRIGGER_STOP:
+			schedule_task_cancel(dev->task);
+			break;
+		}
+	}
+
+	return ret;
+}
+
 /** See comp_ops::trigger */
 static inline int comp_trigger(struct comp_dev *dev, int cmd)
 {
+	int ret;
 	assert(dev->drv->ops.trigger);
 
-	return (dev->is_shared && !cpu_is_me(dev->ipc_config.core)) ?
-		comp_trigger_remote(dev, cmd) : dev->drv->ops.trigger(dev, cmd);
+	if (dev->is_shared && !cpu_is_me(dev->ipc_config.core))
+		ret = comp_trigger_remote(dev, cmd);
+	else
+		ret = comp_trigger_local(dev, cmd);
+
+	return ret;
```

Comment:
btw, there's no reason not to keep the above pattern and just change it to
```
	return (dev->is_shared && !cpu_is_me(dev->ipc_config.core)) ?
		comp_trigger_remote(dev, cmd) : comp_trigger_local(dev, cmd);
```

Reply:
I did it on purpose
It WAY more clear to read and WAY more easy to debug step by step or when setting a breakpoint.

And in result the machine code is 1:1 same.
Couple of more bytes in git repo does not bother me at all